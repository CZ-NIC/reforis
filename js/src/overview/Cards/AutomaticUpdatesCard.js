import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { useAPIGet, withSpinnerOnSending, withErrorMessage } from "foris";
import API_URLs from "../../common/API";

export default function AutomaticUpdates() {
    const [getAutomaticUpdatesResponse, getAutomaticUpdates] = useAPIGet(
        API_URLs.updatesEnabled
    );
    useEffect(() => {
        getAutomaticUpdates();
    }, [getAutomaticUpdates]);

    return (
        <>
            <AutomaticUpdatesCardWithErrorAndSpinner
                apiState={getAutomaticUpdatesResponse.state}
                details={getAutomaticUpdatesResponse.data || {}}
            />
        </>
    );
}

AutomaticUpdatesCard.propTypes = {
    details: PropTypes.object.isRequired,
};

function AutomaticUpdatesCard({ details: { enabled } }) {
    return (
        <div className="col mb-4">
            <div className="card user-select-none">
                <div className="card-body">
                    <div className="row align-items-center">
                        <div className="col">
                            <h6 className="text-uppercase text-muted mb-2 ">
                                Automatic Updates
                            </h6>
                            <span className="h3 mb-0">
                                {enabled ? "Activated" : "Disabled"}
                            </span>
                        </div>
                        <div className="col-auto">
                            <span
                                className={`h2 mb-0 text-${
                                    enabled ? "success" : "danger"
                                }`}
                            >
                                <i
                                    className={`fas fa-${
                                        enabled ? "check" : "times"
                                    }`}
                                />
                            </span>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}

const AutomaticUpdatesCardWithErrorAndSpinner = withSpinnerOnSending(
    withErrorMessage(AutomaticUpdatesCard)
);
