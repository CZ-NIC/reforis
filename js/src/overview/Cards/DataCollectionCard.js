import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { useAPIGet, withSpinnerOnSending, withErrorMessage } from "foris";
import { API_MODULE_URLs } from "../../common/API";


export default function DataCollection() {
    const [getDataCollectionResponse, getDataCollection] = useAPIGet(API_MODULE_URLs.dataCollection);
    useEffect(() => {
        getDataCollection();
    }, [getDataCollection]);

    return (
        <>
            <DataCollectionCardWithErrorAndSpinner
                apiState={getDataCollectionResponse.state}
                details={getDataCollectionResponse.data || {}}
            />

        </>
    );
}

DataCollectionCard.propTypes = {
    details: PropTypes.object.isRequired,
};

function DataCollectionCard({ details: { eula } }) {
    return (
        <div className="col mb-4">
            <div className="card">
                <div className="card-body">
                    <div className="row align-items-center">
                        <div className="col">
                            <h6 className="text-uppercase text-muted mb-2">Data Collection</h6>
                            <span className="h3 mb-0">{eula === 1 ? "Activated" : "Disabled"}</span>
                        </div>
                        <div className="col-auto">
                            <span className={`h2 mb-0 text-${eula === 1 ? "success" : "danger"}`}>
                                <i className="fas fa-check" />
                            </span>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

const DataCollectionCardWithErrorAndSpinner = withSpinnerOnSending(withErrorMessage(DataCollectionCard));

