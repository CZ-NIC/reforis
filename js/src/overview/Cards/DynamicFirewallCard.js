/*
 * Copyright (c) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

DynamicFirewallCard.propTypes = {
    activated: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
        .isRequired,
};

export default function DynamicFirewallCard({ activated }) {
    return (
        <div className="col mb-4">
            <div className="card user-select-none">
                <div className="card-body">
                    <div className="row align-items-center">
                        <div className="col">
                            <h6 className="text-uppercase text-muted mb-2">
                                {_("Dynamic Firewall")}
                            </h6>
                            <span className="status">
                                {activated ? _("Activated") : _("Disabled")}
                            </span>
                        </div>
                        <div className="col-auto">
                            <span
                                className={`h2 text-${
                                    activated ? "success" : "danger"
                                }`}
                            >
                                <i
                                    className={`fas fa-${
                                        activated ? "check" : "times"
                                    }`}
                                />
                            </span>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}
