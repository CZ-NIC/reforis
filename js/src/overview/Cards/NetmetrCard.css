/*
 * Copyright (C) 2021 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

.text-to-bottom {
    position: absolute;
    bottom: 1.25rem;
}
