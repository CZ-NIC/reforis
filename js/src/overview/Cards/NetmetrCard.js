/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { useAPIGet, withSpinnerOnSending, withErrorMessage } from "foris";
import moment from "moment";
import PropTypes from "prop-types";
import { API_MODULE_URLs } from "../../common/API";

export default function Netmetr() {
    const [getDataState, getDataRequest] = useAPIGet(API_MODULE_URLs.netmetr);
    useEffect(() => {
        getDataRequest();
    }, [getDataRequest]);

    return (
        <NetmetrCardWithErrorAndSpinner
            apiState={getDataState.state}
            tests={getDataState.data || {}}
        />
    );
}

NetmetrCard.propTypes = {
    tests: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
};

function NetmetrCard({ tests: { performed_tests: tests } }) {
    const testsCount = typeof tests !== "undefined" ? tests.length : null;
    const lastTest = testsCount > 0 ? tests[0] : null;
    const timeFromNow =
        testsCount > 0
            ? moment(
                  moment.unix(lastTest.time).locale(ForisTranslations.locale)
              ).fromNow()
            : null;
    return (
        <div className="col mb-4">
            <div className="card h-100 user-select-none">
                <div className="card-body">
                    <h6 className="text-uppercase text-muted mb-2">
                        {_("NetMetr")}
                        <Link
                            to={{
                                pathname: "/netmetr/speed-test",
                            }}
                            className="text-secondary"
                            title={_("Go to NetMetr")}
                        >
                            <i className="fas fa-chevron-right float-right" />
                        </Link>
                    </h6>
                    {lastTest != null ? (
                        <>
                            <form>
                                <table className="table table-borderless table-hover offset-lg-3 col-lg-6 col-sm-12">
                                    <tbody>
                                        <tr>
                                            <th scope="row">
                                                {_("Download")}
                                                <span
                                                    className="badge badge-secondary"
                                                    title={_(
                                                        "Megabit per second"
                                                    )}
                                                >
                                                    Mb/s
                                                </span>
                                            </th>
                                            <td className="text-right">
                                                {lastTest.speed_download}
                                            </td>
                                        </tr>
                                        <tr>
                                            <th scope="row">
                                                {_("Upload")}
                                                <span
                                                    className="badge badge-secondary"
                                                    title={_(
                                                        "Megabit per second"
                                                    )}
                                                >
                                                    Mb/s
                                                </span>
                                            </th>
                                            <td className="text-right">
                                                <span>
                                                    {lastTest.speed_upload}
                                                </span>
                                            </td>
                                        </tr>
                                        <tr>
                                            <th scope="row">
                                                {_("Ping")}
                                                <span
                                                    className="badge badge-secondary"
                                                    title={_("Millisecond")}
                                                >
                                                    ms
                                                </span>
                                            </th>
                                            <td className="text-right">
                                                <span>{lastTest.ping}</span>
                                            </td>
                                        </tr>
                                        <tr>
                                            <th scope="row">{_("Link")}</th>
                                            <td className="text-right">
                                                <a
                                                    target="_blank"
                                                    rel="noopener noreferrer"
                                                    href={`https://www.netmetr.cz/en/detail.html?${lastTest.test_uuid}`}
                                                >
                                                    {_("Details")}
                                                    <sup>
                                                        <i className="fas fa-external-link-alt ml-1" />
                                                    </sup>
                                                </a>
                                            </td>
                                        </tr>
                                    </tbody>
                                </table>
                            </form>
                            <p className="card-text">
                                <small className="text-muted">
                                    {_("Performed")}
                                    &nbsp;
                                    {timeFromNow}
                                </small>
                            </p>
                        </>
                    ) : (
                        <p className="text-muted p-2">
                            {_("No tests have been performed lately.")}
                        </p>
                    )}
                </div>
            </div>
        </div>
    );
}

const NetmetrCardWithErrorAndSpinner = withSpinnerOnSending(
    withErrorMessage(NetmetrCard)
);
