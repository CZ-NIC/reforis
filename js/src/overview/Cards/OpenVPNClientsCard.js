import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { useAPIGet, withSpinnerOnSending, withErrorMessage } from "foris";
import { API_MODULE_URLs } from "../../common/API";

export default function OpenVPNClients() {
    const [getOpenVPNClientsResponse, getOpenVPNClients] = useAPIGet(
        API_MODULE_URLs.openvpnClients
    );
    useEffect(() => {
        getOpenVPNClients();
    }, [getOpenVPNClients]);

    return (
        <>
            <OpenVPNClientsCardWithErrorAndSpinner
                apiState={getOpenVPNClientsResponse.state}
                clients={getOpenVPNClientsResponse.data || {}}
            />
        </>
    );
}

OpenVPNClients.propTypes = {
    clients: PropTypes.object.isRequired,
};

function OpenVPNClientsCard({ clients }) {
    console.log(clients);
    return (
        <div className="col mb-4">
            <div className="card h-100">
                <div className="card-body">
                    <h6 className="text-uppercase text-muted mb-2">
                        OpenVPN Clients
                        <a href="javasript;" className="text-secondary">
                            <i className="fas fa-cog float-right" />
                        </a>
                    </h6>
                    <form>
                        <table className="table table-borderless table-hover offset-lg-3 col-lg-6 col-sm-12">
                            <tbody>
                                {clients
                                    .slice(0, 5)
                                    .filter(
                                        (client) =>
                                            client.status === "revoked" ||
                                            client.status === "valid"
                                    )
                                    .map((client) => (
                                        <tr key={client.id}>
                                            <th scope="row">
                                                <span>{client.name}</span>
                                            </th>
                                            <td style={{ textAlign: "center" }}>
                                                <span
                                                    className={`text-${
                                                        client.status ===
                                                        "valid"
                                                            ? "success"
                                                            : "warning"
                                                    }`}
                                                >
                                                    <i
                                                        className={`fas fa-${
                                                            client.status ===
                                                            "valid"
                                                                ? "check"
                                                                : "exclamation-triangle"
                                                        }`}
                                                    />
                                                </span>
                                            </td>
                                        </tr>
                                    ))}
                            </tbody>
                        </table>
                    </form>
                </div>
            </div>
        </div>
    );
}

const OpenVPNClientsCardWithErrorAndSpinner = withSpinnerOnSending(
    withErrorMessage(OpenVPNClientsCard)
);
