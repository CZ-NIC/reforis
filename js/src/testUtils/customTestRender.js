/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

/* eslint import/export: "off" */

import React from 'react';
import {UIDReset} from 'react-uid';
import {StaticRouter} from 'react-router';
import {render} from '@testing-library/react'

function Wrapper({children}) {
    return <StaticRouter>
        <UIDReset>
            {children}
        </UIDReset>
    </StaticRouter>
}

const customTestRender = (ui, options) => render(ui, {wrapper: Wrapper, ...options});

// re-export everything
export * from '@testing-library/react'

// override render method
export {customTestRender as render}
