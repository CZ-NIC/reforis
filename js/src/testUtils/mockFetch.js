/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

export default function mockFetch(data) {
    return jest.fn(() =>
        new Promise((resolve, reject) => {
            resolve({
                ok: true,
                json: () => {
                    return data ? data : {};
                },
            });
        })
    );
}
