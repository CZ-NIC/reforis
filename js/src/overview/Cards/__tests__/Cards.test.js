/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import mockAxios from "jest-mock-axios";
import { render, wait, getByText } from "foris/testUtils/customTestRender";

import AutomaticUpdatesCard from "../AutomaticUpdatesCard";
import DataCollectionCard from "../DataCollectionCard";
import DynamicFirewallCard from "../DynamicFirewallCard";
import NetmetrCard from "../NetmetrCard";
import OpenVPNClientsCard from "../OpenVPNClientsCard";

import {
    automaticUpdatesCardFixture,
    automaticUpdatesCardFixture2,
    dataCollectionCardFixture,
    dataCollectionCardFixture2,
    netmetrCardFixture,
    netmetrCardFixture2,
    openVPNClientsCardFixture,
    openVPNClientsCardFixture2,
} from "../../__tests__/__fixtures__/overview";

describe("<Cards/>", () => {
    describe("<AutomaticUpdatesCard/>", () => {
        it("Snapshot: activated", async () => {
            const { container } = render(<AutomaticUpdatesCard />);
            mockAxios.mockResponse({ data: automaticUpdatesCardFixture });
            await wait(() => getByText(container, "Automatic Updates"));

            expect(container).toMatchSnapshot();
        });

        it("Snapshot: disabled", async () => {
            const { container } = render(<AutomaticUpdatesCard />);
            mockAxios.mockResponse({ data: automaticUpdatesCardFixture2 });
            await wait(() => getByText(container, "Automatic Updates"));

            expect(container).toMatchSnapshot();
        });
    });

    describe("<DataCollectionCard/>", () => {
        it("Snapshot: activated", async () => {
            const { container } = render(<DataCollectionCard />);
            mockAxios.mockResponse({ data: dataCollectionCardFixture });
            await wait(() => getByText(container, "Data Collection"));

            expect(container).toMatchSnapshot();
        });

        it("Snapshot: disabled", async () => {
            const { container } = render(<DataCollectionCard />);
            mockAxios.mockResponse({ data: dataCollectionCardFixture2 });
            await wait(() => getByText(container, "Data Collection"));

            expect(container).toMatchSnapshot();
        });
    });

    describe("<DynamicFirewallCard/>", () => {
        const activated = () => true;
        const disabled = () => false;

        it("Snapshot: activated", () => {
            const { container } = render(
                <DynamicFirewallCard activated={activated()} />
            );
            expect(container).toMatchSnapshot();
        });

        it("Snapshot: disabled", () => {
            const { container } = render(
                <DynamicFirewallCard activated={disabled()} />
            );
            expect(container).toMatchSnapshot();
        });
    });

    describe("<NetmetrCard/>", () => {
        it("Snapshot: contains tests", async () => {
            const { container } = render(<NetmetrCard />);
            mockAxios.mockResponse({ data: netmetrCardFixture });
            await wait(() => getByText(container, "NetMetr"));

            expect(container).toMatchSnapshot();
        });

        it("Snapshot: empty", async () => {
            const { container } = render(<NetmetrCard />);
            mockAxios.mockResponse({ data: netmetrCardFixture2 });
            await wait(() => getByText(container, "NetMetr"));

            expect(container).toMatchSnapshot();
        });
    });

    describe("<OpenVPNClientsCard/>", () => {
        it("Snapshot: contains clients", async () => {
            const { container } = render(<OpenVPNClientsCard />);
            mockAxios.mockResponse({ data: openVPNClientsCardFixture });
            await wait(() => getByText(container, "OpenVPN Clients"));

            expect(container).toMatchSnapshot();
        });

        it("Snapshot: empty", async () => {
            const { container } = render(<OpenVPNClientsCard />);
            mockAxios.mockResponse({ data: openVPNClientsCardFixture2 });
            await wait(() => getByText(container, "OpenVPN Clients"));

            expect(container).toMatchSnapshot();
        });
    });
});
