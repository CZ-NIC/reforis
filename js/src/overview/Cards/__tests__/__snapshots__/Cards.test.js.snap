// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards/> <AutomaticUpdatesCard/> Snapshot: activated 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card user-select-none"
    >
      <div
        class="card-body"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col"
          >
            <h6
              class="text-uppercase text-muted mb-2 "
            >
              Automatic Updates
            </h6>
            <span
              class="status"
            >
              Activated
            </span>
          </div>
          <div
            class="col-auto"
          >
            <span
              class="h2 text-success"
            >
              <i
                class="fas fa-check"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cards/> <AutomaticUpdatesCard/> Snapshot: disabled 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card user-select-none"
    >
      <div
        class="card-body"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col"
          >
            <h6
              class="text-uppercase text-muted mb-2 "
            >
              Automatic Updates
            </h6>
            <span
              class="status"
            >
              Disabled
            </span>
          </div>
          <div
            class="col-auto"
          >
            <span
              class="h2 text-danger"
            >
              <i
                class="fas fa-times"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cards/> <DataCollectionCard/> Snapshot: activated 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card user-select-none"
    >
      <div
        class="card-body"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col"
          >
            <h6
              class="text-uppercase text-muted mb-2"
            >
              Data Collection
            </h6>
            <span
              class="status"
            >
              Activated
            </span>
          </div>
          <div
            class="col-auto"
          >
            <span
              class="h2 text-success"
            >
              <i
                class="fas fa-check"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cards/> <DataCollectionCard/> Snapshot: disabled 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card user-select-none"
    >
      <div
        class="card-body"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col"
          >
            <h6
              class="text-uppercase text-muted mb-2"
            >
              Data Collection
            </h6>
            <span
              class="status"
            >
              Disabled
            </span>
          </div>
          <div
            class="col-auto"
          >
            <span
              class="h2 text-danger"
            >
              <i
                class="fas fa-times"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cards/> <DynamicFirewallCard/> Snapshot: activated 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card user-select-none"
    >
      <div
        class="card-body"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col"
          >
            <h6
              class="text-uppercase text-muted mb-2"
            >
              Dynamic Firewall
            </h6>
            <span
              class="status"
            >
              Activated
            </span>
          </div>
          <div
            class="col-auto"
          >
            <span
              class="h2 text-success"
            >
              <i
                class="fas fa-check"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cards/> <DynamicFirewallCard/> Snapshot: disabled 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card user-select-none"
    >
      <div
        class="card-body"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col"
          >
            <h6
              class="text-uppercase text-muted mb-2"
            >
              Dynamic Firewall
            </h6>
            <span
              class="status"
            >
              Activated
            </span>
          </div>
          <div
            class="col-auto"
          >
            <span
              class="h2 text-success"
            >
              <i
                class="fas fa-check"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cards/> <NetmetrCard/> Snapshot: contains tests 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card h-100 user-select-none"
    >
      <div
        class="card-body"
      >
        <h6
          class="text-uppercase text-muted mb-2"
        >
          NetMetr
          <a
            class="text-secondary"
            href="/netmetr/speed-test"
            title="Go to NetMetr"
          >
            <i
              class="fas fa-chevron-right float-right"
            />
          </a>
        </h6>
        <form>
          <table
            class="table table-borderless table-hover offset-lg-3 col-lg-6 col-sm-12"
          >
            <tbody>
              <tr>
                <th
                  scope="row"
                >
                  Download
                  <span
                    class="badge badge-secondary"
                    title="Megabit per second"
                  >
                    Mb/s
                  </span>
                </th>
                <td
                  class="text-right"
                >
                  930
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Upload
                  <span
                    class="badge badge-secondary"
                    title="Megabit per second"
                  >
                    Mb/s
                  </span>
                </th>
                <td
                  class="text-right"
                >
                  <span>
                    930
                  </span>
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Ping
                  <span
                    class="badge badge-secondary"
                    title="Millisecond"
                  >
                    ms
                  </span>
                </th>
                <td
                  class="text-right"
                >
                  <span>
                    -1
                  </span>
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Link
                </th>
                <td
                  class="text-right"
                >
                  <a
                    href="https://www.netmetr.cz/en/detail.html?c7a6f82d-5128-47f3-a3f9-d7e9d9af64bd"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Details
                    <sup>
                      <i
                        class="fas fa-external-link-alt ml-1"
                      />
                    </sup>
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </form>
        <p
          class="card-text"
        >
          <small
            class="text-muted"
          >
            Performed
            Â 
            in 2 years
          </small>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cards/> <NetmetrCard/> Snapshot: empty 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card h-100 user-select-none"
    >
      <div
        class="card-body"
      >
        <h6
          class="text-uppercase text-muted mb-2"
        >
          NetMetr
          <a
            class="text-secondary"
            href="/netmetr/speed-test"
            title="Go to NetMetr"
          >
            <i
              class="fas fa-chevron-right float-right"
            />
          </a>
        </h6>
        <p
          class="text-muted p-2"
        >
          No tests have been performed lately.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cards/> <OpenVPNClientsCard/> Snapshot: contains clients 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card h-100 user-select-none"
    >
      <div
        class="card-body"
      >
        <h6
          class="text-uppercase text-muted mb-2"
        >
          OpenVPN Clients
          <a
            class="text-secondary"
            href="/openvpn/client-settings"
            title="Go to OpenVPN Client Settings"
          >
            <i
              class="fas fa-chevron-right float-right"
            />
          </a>
        </h6>
        <form>
          <table
            class="table table-borderless table-hover offset-lg-3 col-lg-6 col-sm-12"
          >
            <tbody>
              <tr>
                <th
                  scope="row"
                >
                  <span>
                    turris
                  </span>
                </th>
                <td
                  class="text-right"
                >
                  <span
                    class="text-success"
                  >
                    <i
                      class="fas fa-check"
                    />
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cards/> <OpenVPNClientsCard/> Snapshot: empty 1`] = `
<div>
  <div
    class="col mb-4"
  >
    <div
      class="card h-100 user-select-none"
    >
      <div
        class="card-body"
      >
        <h6
          class="text-uppercase text-muted mb-2"
        >
          OpenVPN Clients
          <a
            class="text-secondary"
            href="/openvpn/client-settings"
            title="Go to OpenVPN Client Settings"
          >
            <i
              class="fas fa-chevron-right float-right"
            />
          </a>
        </h6>
        <p
          class="text-muted p-2"
        >
          There are no clients added yet.
        </p>
      </div>
    </div>
  </div>
</div>
`;
